jax/lax.py

```diff
@@ -2233,12 +2232,63 @@ def _scatter_transpose_rule(t, operand, scatter_indices, updates,
                       slice_sizes=slice_sizes)
   return [operand_t, None, update_t]
 
+def _scatter_batching_rule(batched_args, batch_dims, update_jaxpr,
+                           update_consts, dimension_numbers, updates_shape):
+  operand, scatter_indices, updates = batched_args
+  operand_bdim, scatter_indices_bdim, updates_bdim = batch_dims
+  del update_jaxpr, update_consts, updates_shape  # Unused.
+
+  size = next(x.shape[ax] for x, ax in zip(batched_args, batch_dims)
+              if ax is not None)
+  operand = batching.bdim_at_front(operand, operand_bdim, broadcast_size=size,
+                                   force_broadcast=True)
+  operand_bdim = 0
+
+  if scatter_indices_bdim is not None and updates_bdim is None:
+    raise NotImplementedError  # TODO(mattjj,phawkins)
```

Comment:
Maybe do this too, given you already did the harder case?

Reply:
Nah, I want to move on to other things and it's not clear that we care about this other case (which can't be generated by vmap+grad on gather, but only by vmapping over lax.scatter_add directly).